### PR TITLE
Automate Publish + Release

### PR DESCRIPTION
**Pull request by alxjrvs**
## TL;DR

Just merge a `package.json` version bump to main and a github workflow will do the rest. 

"the rest":  publish the new version on NPM, make a `git` release (with binary), and open a PR to our `homebrew-gnar` repo with updated changes. 

## Longer Read 

This PR enables a certain amount of automatic publishing (to the NPM marketplace) releasing (in Github, with Tags + Changelog) in this repo, and opening a PR in the `homebrew-gnar` repo.

All that is required to deploy the app is for `main` to receive a push that contains a bump to the `package.json` versions file. 

When we merge such a commit in (probably from a PR!) a few different things kick off in that workflow. See [Publishing.md](https://github.com/TheGnarCo/gnar-cli/blob/d40131e9e2f663d8652044f42c80263786f41da3/PUBLISHING.md) for more in-depth info. 

The three biggest outputs are the NPM update, the Git Release, and the Homebrew PR. 

[Here is an example](https://github.com/TheGnarCo/gnar-cli/releases/tag/test-v0.0.12) of what one of these git releases looks like.

[Here is an example](https://github.com/TheGnarCo/homebrew-gnar/pull/8) of a homebrew-gnar PR. (NOTE: when automatically generated, these PR's are not in draft mode, and they do not include "DO NOT MERGE". )

Both were generated using a `dry-run` of the NPM publish action with a prefix `test-` that would not be present in the final version of the script. 

### Also 

I changed the name of the jobs on the `lint` and `docs` job from `build` to - you guessed it - `lint` and `docs`.

### CODE REVIEW

File: PUBLISHING.md [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+The NPM version page can be found [here](https://www.npmjs.com/package/gnar-cli), and the repo containing the Homebrew Formula is [here](https://github.com/TheGnarCo/homebrew-gnar).
+
+### Publishing a new Version
+
+This repo uses a Github Workflow, `publish.yml`, to handle most of the tasks automatically.
+
+In order to publish using the Github Workflow:
+
+1. Create a pull request where you incremenet the `package.json` version. (This doesn't have to be the only thing in the PR, but I do recommend it be the only thing in the PR)
+1. Merge this PR in to main. (Pushing this commit to main is what triggers the `publish` workflow. It won't get very far if there isn't an update to `package.json`. TODO: Make this cancel out earlier so we don't build and run tests!)
```
Comment (WMahoney09) [inline PUBLISHING.md:16]: > TODO: Make this cancel out earlier so we don't build and run tests!)

do we have a ticket or issue or some such to track this?

**Review by WMahoney09 — APPROVED** [review]
:shipit:

File: PUBLISHING.md (hunk 1 above)
Comment (alxjrvs) [reply PUBLISHING.md:16]: Nope! I can make one though.

Comment (alxjrvs) [reply PUBLISHING.md:16]: https://github.com/TheGnarCo/gnar-cli/issues/45

Comment (WMahoney09) [reply PUBLISHING.md:16]: 🎱 should we remove the todo from this `md`?
